packages/components/src/components/PasswordCreationField/components/TogglePasswordVisibilityButton/TogglePasswordVisibilityButton.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { type FC } from "react";
+import { Tooltip, TooltipTrigger } from "@/components/Tooltip";
+import Button from "@/components/Button";
+import { IconHide, IconShow } from "@/components/Icon/components/icons";
+import { Action, type ActionFn } from "@/components/Action";
+import locales from "./../../locales/*.locale.json";
+import { useLocalizedStringFormatter } from "react-aria";
+
+interface Props {
+  isDisabled?: boolean;
+  isVisible: boolean;
+  onPress: ActionFn;
+}
+
+/** @internal */
+export const TogglePasswordVisibilityButton: FC<Props> = ({
+  isVisible,
+  isDisabled = false,
+  onPress,
+}) => {
+  const translate = useLocalizedStringFormatter(locales);
+
+  return (
+    <Action action={onPress}>
+      <TooltipTrigger>
+        <Button
+          size="m"
+          variant="plain"
+          color="secondary"
+          isDisabled={isDisabled}
+          data-component="toggleRevealPassword"
+        >
+          {!isVisible ? (
+            <>
+              <IconShow />
+              <Tooltip>
+                {translate.format("button.generate.tooltip.show")}
+              </Tooltip>
+            </>
+          ) : (
+            <>
+              <IconHide />
+              <Tooltip>
+                {translate.format("button.generate.tooltip.hide")}
+              </Tooltip>
+            </>
+          )}
+        </Button>
+      </TooltipTrigger>
+    </Action>
+  );
```

Comment:
_Disclaimer: This is not (only) my personal opinion of coding style._

In this package we (even) try to make the returned JSX component structure easy to read (e.g. no large ternary statements). If things becoming complex, parts of the tree can be separated into own variables.

For example the icon and tooltip text can be extracted like this.

```suggestion
  const icon = isVisible ? <IconHide /> : <IconHide />;
  const tooltipText = translate.format(
    "button.generate.tooltip." + (isVisible ? "hide" : "show"),
  );

  return (
    <Action action={onPress}>
      <TooltipTrigger>
        <Button
          size="m"
          variant="plain"
          color="secondary"
          isDisabled={isDisabled}
          data-component="toggleRevealPassword"
        >
          {icon}
          <Tooltip>{tooltipText}</Tooltip>
        </Button>
      </TooltipTrigger>
    </Action>
  );
```